transactor/src/main/scala/io/mediachain/transactor/CborSerialization.scala

```diff
@@ -0,0 +1,452 @@
+
+  import scala.util.Try
+
+  /**
+    * Try to deserialize a `DataObject` from a cbor `CValue`
+    * @param cValue the `CValue` to decode
+    * @return a `DataObject`, or `DeserializationError` on failure
+    */
+  def dataObjectFromCbor(cValue: CValue): Xor[DeserializationError, DataObject] =
+    fromCbor(cValue) match {
+      case Xor.Left(err) => Xor.left(err)
+      case Xor.Right(dataObject: DataObject) => Xor.right(dataObject)
+      case Xor.Right(unknownObject) =>
+        Xor.left(UnexpectedCborType(
+          s"Expected DataObject, but got ${unknownObject.getClass.getTypeName}"
+        ))
+    }
+
+  /**
+    * Try to deserialize a `JournalEntry` from a cbor `CValue`
+    * @param cValue the `CValue` to decode
+    * @return a `JournalEntry`, or DeserializationError on failure
+    */
+  def journalEntryFromCbor(cValue: CValue): Xor[DeserializationError, JournalEntry] =
+    fromCbor(cValue) match {
+      case Xor.Left(err) => Xor.left(err)
+      case Xor.Right(journalEntry: JournalEntry) => Xor.right(journalEntry)
+      case Xor.Right(unknownObject) =>
+        Xor.left(UnexpectedCborType(
+          s"Expected JournalEntry, but got ${unknownObject.getClass.getTypeName}"
+        ))
+    }
+
+  /**
+    * Try to deserialize some `CborSerializable` object from a byte array.
+    * @param bytes an array of (presumably) cbor-encoded data
+    * @return a `CborSerializable` object, or a `DeserializationError` on failure
+    */
+  def fromCborBytes(bytes: Array[Byte]): Xor[DeserializationError, CborSerializable] =
+    CborCodec.decode(bytes) match {
+      case (_: CTag) :: (taggedValue: CValue) :: _ => fromCbor(taggedValue)
+      case (cValue: CValue) :: _ => fromCbor(cValue)
+      case Nil => Xor.left(CborDecodingFailed())
+    }
+
+  /**
+    * Try to deserialize a `CborSerializable` object from a cbor `CValue`
+    * @param cValue the `CValue` to decode
+    * @return a `CborSerializable` object, or a `DeserializationError` on failure
+    */
+  def fromCbor(cValue: CValue): Xor[DeserializationError, CborSerializable] =
+    cValue match {
+      case (cMap: CMap) => fromCMap(cMap)
+      case _ => Xor.left(UnexpectedCborType(
+        s"Expected CBOR map, but received ${cValue.getClass.getName}"
+      ))
+    }
+
+  /**
+    * Try to deserialize some `CborSerializable` object from a cbor `CMap`.
+    * @param cMap a cbor map representing the object to decode
+    * @return a `CborSerializable` object, or a `DeserializationError` on failure
+    */
+  def fromCMap(cMap: CMap)
+  : Xor[DeserializationError, CborSerializable] = {
+
+    // TODO: create multiple deserializer maps for different contexts
+    // e.g. DataStore should deserialize chain cells to specific subtypes, etc
+    val transactorDeserializers
+    : Map[String, CborDeserializer[CborSerializable]] = Map(
+      CBORTypeNames.Entity -> EntityDeserializer
+        .asInstanceOf[CborDeserializer[CborSerializable]],
```

Comment:
ah, that Seq / map trick looks nice :)

good call about moving to the top level